Sources/NIOFileSystem/FileSystem.swift

```diff
@@ -438,29 +438,37 @@ public struct FileSystem: Sendable, FileSystemProtocol {
         }
     }
 
-    @discardableResult
-    private func removeItemSequentially(
-        at path: FilePath
-    ) async throws -> Int {
-        var (subdirectories, filesRemoved) = try await self.withDirectoryHandle(
+    func _collectItemsInDirectory(at path: FilePath) async throws -> ([FilePath], [FilePath]) {
```

Comment:
Can you add labels to the tuples? It's too easy to mix them up otherwise.

Reply:
Actually, do we even need to split them? We could just return the directory entries and the caller can then iterate over them and do as they need. This would avoid allocating an array unnecessarily as well (not that I think it'll have any meaningful impact on perf).

Reply:
I can get behind adding labels. If we only return a single array, then the callers would need to perform the sorting themselves, and thus replicate the bulk of what this utility does. I'm not wed to this existing, but I think if we have it then might as well use it to remove some level of duplication?

Reply:
Had a quick discussion offline. Reverted the addition of `_collectItemsInDirectory` in [e9df08e](https://github.com/apple/swift-nio/pull/3008/commits/e9df08e1c69778b5afeb743d5dc050b507a62d3b)